### PR TITLE
fix: point 'main' and 'types' to dist folder to fix import targets for local development

### PR DESCRIPTION
**Pull request by benjroy**
patches the definition of main and types in package.json to be consistent for local development and what is published.

## Change description

Why am I making this change?

I've been following the guide for getting our backstage dev environment set up.  I kept running into some weird behavior where the first time i `yalc add ....` the backstage-plugin dep into the backstage repo, it worked, but would fail everytime after i did a `yalc push` in the backstage plugin to update the code.

on further inspection I could see that yalc was dragging over `src/index.ts` even if `src/` was added to `.npmignore`.  `npm pack` was not dragging the file into the `gz` archive of the package.  I was getting confused.

Eventually I noticed these fields in package.json pointing towards `src/index.ts`.  Those should always point to the transpiled files in `dist`, and making that change fixed the weird error where yalc was dragging in the src file and only that src file.  it would barf trying to pull in all the other files in src directory that `src/index` referenced.  (maybe yalc ignored the `publishConfig` field entirely... seems like a likely bug)

A side effect of this change for the local workflow with this plugin is that we will need to be constantly building to `dist` with a watcher while we work.  instead of relying on hot-reloading directly from changes in src file while working in the `backstage` repo.

TL;DR;  `@cortexapps/backstage-plugin` is meant to be installed as a dependency, so keep that consistent between local development vs using the public package.

## Type of change

- [x] Bug fix (fixes an issue)
- [ ] New feature (adds functionality)
- [ ] Improvement (improves codebase without changing functionality)

### Development

- [ ] Lint rules pass locally
- [ ] Application changes have been tested thoroughly
- [ ] Automated tests covering modified code pass

### Security

- [ ] Security impact of change has been considered
- [ ] Code follows company security practices and guidelines

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,14 +1,12 @@
 {
   "name": "@cortexapps/backstage-plugin",
   "version": "1.2.9",
```
Comment (nikhilunni) [inline package.json:3]: i would bump the lib version here, so we can test that the new published package will still work as expected

**Review by nikhilunni — DISMISSED** [review]
looks good!

File: package.json (hunk 1 above)
Comment (benjroy) [reply package.json:3]: Re: version: Sounds good!  Do we manually publish these packages, or are we luck that CI publish for us when it sees a version change?

**Comment by benjroy** [comment]
Ok, I also added a `dev` command to package.json that will watch changes in src directory and then run the tst && build && yalc push.

Workflow should be to come to this repo and run `yarn dev` and start editing source code.

**Comment by benjroy** [comment]
I'm going to back out the change to the `main` entry in package.json and keep it pointing to `src`.  I'll push a commit to have `npm pack` pull in the `src` directory into the shipped archive.  This will allow us to avoid stack traces like this:

```
Error: i errored here
    at SettingsPage (webpack-internal:///../../.yalc/@cortexapps/backstage-plugin/dist/esm/index-b240728b.esm.js:149:9)
    at renderWithHooks (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:14986:18)
    at mountIndeterminateComponent (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:17812:13)
    at beginWork (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:19050:16)
    at HTMLUnknownElement.callCallback (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:3946:14)
    at Object.invokeGuardedCallbackDev (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:3995:16)
    at invokeGuardedCallback (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:4057:31)
    at beginWork$1 (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:23960:7)
    at performUnitOfWork (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:22772:12)
    at workLoopSync (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:22703:5)
```

in favor of ones with more useful lines while we are developing:

```
Error: i errored here
    at SettingsPage (webpack-internal:///../../.yalc/@cortexapps/backstage-plugin/src/components/SettingsPage/SettingsPage.tsx:31:9)
    at renderWithHooks (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:14986:18)
    at mountIndeterminateComponent (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:17812:13)
    at beginWork (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:19050:16)
    at HTMLUnknownElement.callCallback (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:3946:14)
    at Object.invokeGuardedCallbackDev (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:3995:16)
    at invokeGuardedCallback (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:4057:31)
    at beginWork$1 (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:23960:7)
    at performUnitOfWork (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:22772:12)
    at workLoopSync (webpack-internal:///../../node_modules/@hot-loader/react-dom-v17/cjs/react-dom.development.js:22703:5)
```

which isn't exactly where it is, but at least it is getting us inside the right file:

image

File: package.json (hunk 1 above)
Comment (nikhilunni) [reply package.json]: CI's got us 😄

File: package.json [hunk 2]
```diff
@@ -26,7 +26,17 @@
     "diff": "backstage-cli plugin:diff",
     "prepack": "backstage-cli package prepack",
     "postpack": "backstage-cli package postpack",
-    "clean": "backstage-cli package clean"
+    "clean": "backstage-cli package clean",
+    "dev": "yarn watch",
+    "watch": "npm-watch yalc-push",
+    "yalc-push": "yarn build && yalc push"
+  },
+  "watch": {
```
Comment (nikhilunni) [inline package.json:34]: this is amazing 🔥

**Review by nikhilunni — DISMISSED** [review]
looks great!!

File: package.json [hunk 3]
```diff
@@ -49,6 +59,7 @@
     "lodash": "^4.17.21",
     "material-table": "^1.69.2",
     "moment": "^2.29.1",
+    "npm-watch": "^0.11.0",
```
Comment (benjroy) [inline package.json:62]: should definitely be a devDependency.  pushing up a patch before merging

**Review by nikhilunni — APPROVED** [review]
